nginx.conf

```diff
@@ -10,7 +10,10 @@ events {
 http {
   log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                     '$status $body_bytes_sent "$http_referer" '
-                    '"$http_user_agent" "$http_x_forwarded_for"';
+                    '"$http_user_agent" "$http_x_forwarded_for" "$upstream_cache_status"';
+
+  proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=graphqlsha_requests:10m max_size=${CACHE_MAX_SIZE} inactive=60m use_temp_path=off;
```

Comment:
* `inactive`: The current state allows for `CACHE_TTL` to be greater that `inactive`, so we could have cache misses (if i understand things correctly). I guess that `inactive` should be at least equal and generally higher than `CACHE_TTL`, correct? We could make it configurable but set a reasonable default based on the ttl.
* Where does the `10m` keys come from? At least, I'd make it configurable.

Reply:
-  `inactive` is basically making the cache a effectively LRU cache. Given the nature of our integrations running in a polling fashion and our longest running integration quay-mirror takes ~2000 sec , it made sense to set `inactive` to 60m. So as long as the cache key used within the `inactive` period, we should not see cache misses.
- From https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_path 

> One megabyte zone can store about 8 thousand keys 
So using 10m should give us 80000 keys, which seems more than sufficient. Considering as a general rule of 1 graphql query per integration ( exceptions apply ofcourse) this seems reasonable to keep it hard-coded for now.

Reply:
From the documentation you link:

* _"Cached data that are not accessed during the time specified by the `inactive` parameter get removed from the cache regardless of their freshness. By default, inactive is set to 10 minutes."_
_* `proxy_cache_valid`:  "Sets caching time for different response codes"_ 

That means to me that expiration time is set via `proxy_cache_valid`, the directive you're using below associated with the CACHE_TTL.

This is consistent with other users interpretation too, see links in https://stackoverflow.com/questions/64151378/nginx-cache-inactive-vs-proxy-cache-valid

Reply:
Regarding the size, it never hurts to have configuration parameters, but YMMV :-D

Reply:
Based on https://github.com/app-sre/nginx-gate/pull/10#discussion_r1625559753 we can leave this as it is. I would add the time to delete objects in the cache and the size of the keys zone as parameters though.

Reply:
Made the parameters configurable as you suggested.